File: reorder_sections.py

import os

file_path = r"c:\Users\owner\Downloads\やすらぎの家readdyコード\app\page.tsx"

with open(file_path, "r", encoding="utf-8") as f:
    lines = f.readlines()

greeting_idx = -1
block_start_idx = -1
recruit_idx = -1

for i, line in enumerate(lines):
    if "{/* ごあいさつ */}" in line:
        greeting_idx = i
    elif "{/* 施設紹介ヘッダー */}" in line:
        block_start_idx = i
    elif "{/* 採用情報 */}" in line:
        recruit_idx = i

if greeting_idx == -1 or block_start_idx == -1 or recruit_idx == -1:
    print(f"Error: Could not find all markers. Greeting: {greeting_idx}, Block: {block_start_idx}, Recruit: {recruit_idx}")
    exit(1)

print(f"Markers found: Greeting={greeting_idx}, BlockStart={block_start_idx}, Recruit={recruit_idx}")

# Construct new content
# 1. Everything before Greeting
part1 = lines[:greeting_idx]
# 2. Facility Block (From Intro to just before Recruit)
# Note: recruit_idx puts it at the line "{/* 採用情報 */}". We want all lines before that.
# Assuming formatting: empty lines might be captured in the slice.
part2_block = lines[block_start_idx:recruit_idx]
# 3. Middle (From Greeting to just before Facility Intro)
part3_middle = lines[greeting_idx:block_start_idx]
# 4. Rest (From Recruit to end)
part4 = lines[recruit_idx:]

new_lines = part1 + part2_block + part3_middle + part4

with open(file_path, "w", encoding="utf-8") as f:
    f.writelines(new_lines)

print("Successfully reordered sections.")

File: reorder_sections.js

const fs = require('fs');
const path = 'app/page.tsx';

try {
    const content = fs.readFileSync(path, 'utf8');
    // Detect line ending
    const isCRLF = content.includes('\r\n');
    const splitChar = isCRLF ? '\r\n' : '\n';

    const lines = content.split(splitChar);

    let greetingIdx = -1;
    let blockStartIdx = -1;
    let recruitIdx = -1;

    for (let i = 0; i < lines.length; i++) {
        if (lines[i].includes("{/* ごあいさつ */}")) greetingIdx = i;
        if (lines[i].includes("{/* 施設紹介ヘッダー */}")) blockStartIdx = i;
        if (lines[i].includes("{/* 採用情報 */}")) recruitIdx = i;
    }

    if (greetingIdx === -1 || blockStartIdx === -1 || recruitIdx === -1) {
        console.error(`Markers not found: Greeting=${greetingIdx}, Block=${blockStartIdx}, Recruit=${recruitIdx}`);
        // Fallback search
        console.log("Dumping first 10 lines containing identifiers:");
        lines.forEach((l, i) => {
            if (l.includes('{/*')) console.log(`${i}: ${l}`);
        });
        process.exit(1);
    }

    console.log(`Indices: Greeting=${greetingIdx}, BlockStart=${blockStartIdx}, Recruit=${recruitIdx}`);

    const part1 = lines.slice(0, greetingIdx);
    const block = lines.slice(blockStartIdx, recruitIdx);
    const middle = lines.slice(greetingIdx, blockStartIdx);
    const part4 = lines.slice(recruitIdx);

    const newLines = [...part1, ...block, ...middle, ...part4];
    const newContent = newLines.join(splitChar);

    fs.writeFileSync(path, newContent, 'utf8');
    console.log("Successfully reordered sections.");

} catch (e) {
    console.error(e);
    process.exit(1);
}
